src/icrar/leap-accelerate/algorithm/cuda/PhaseRotate.cu

```diff
@@ -132,15 +137,33 @@ namespace cuda
         deviceMetadata.ToHost(hostMetadata);
         
         auto avg_data_angles = hostMetadata.avg_data.unaryExpr([](std::complex<double> c) -> Radians { return std::arg(c); });
-        auto& indexes = hostMetadata.GetI1();
-        auto avg_data_t = avg_data_angles(indexes, 0); // 1st pol only
+        Eigen::VectorXi indices1 = hostMetadata.GetI1();
+        for(int& v : indices1)
+        {
+            if(v < 0)
+            {
+                v += avg_data_angles.rows(); /// -behaviour in python is to wrap around
+                //TODO: reference antenna should be included, set to 0?
+            }
+        }
```

Comment:
I currently question whether this behaviour was intended, however I suspect there are plans to make parts of this library public for new executables so it may be beneficial to make this safely handle negative indexes or use an unsigned type. I'll draft a utility function up anyway as it may be useful if I encounter negatives again.